model/DataObject.php

```diff
@@ -1575,6 +1575,11 @@ public function getComponents($componentName, $filter = "", $sort = "", $join =
 			);
 		}
 
+		if($filter !== null || $sort !== null || $limit !== null) {
+			Deprecation::notice('4.0', 'The $filter, $sort and $limit parameters for DataObject::getComponents() 
```

Comment:
This needs to be `'3.2'` not `4.0` we'd like to remove it in 4 so the warning should show sooner than that